### ISSUE TITLE
Application logout takes more than 10 seconds sometimes

### ISSUE THREAD

**Issue by anastasiyaig** [status-im/status-desktop#13013]
### Description

![image](https://github.com/status-im/status-desktop/assets/82375995/10c2a3c2-f2a4-4df5-8618-7515c21e033b)

https://ci.status.im/job/status-desktop/job/systems/job/linux/job/x86_64/job/tests-e2e-new/124/allure/#suites/85aa51e65eb2606a1f89f1dfcae42139/8ba337a35d9d993b/

Sometimes autotest encounters that logging out from the app takes longer than 10 seconds. I am not sure which timeout is acceptable though. In app logs i clearly see the logging out was initiated

```
tid=1703015 file=core.nim:27 rpc_method=wallet_getOwnedCollectiblesAsync
INF 2023-12-14 04:36:32.267+00:00 shutting down...                           topics="status-app" tid=1703015 file=nim_status_client.nim:173
INF 2023-12-14 04:36:32.267+00:00 logging out...                             topics="app-controller" tid=1703015 file=app_controller.nim:288
```

However the verification i do (checking if process no longer exists within 10 seconds) sometimes fails, which means the process is still running.

I need a hand here to investigate and perhaps define applicable timeframe for this action. It looks like regression to me now

**Comment by anastasiyaig** [comment]
@igor-sirotin maybe u can help with that? Not sure whom to ping :)

**Comment by anastasiyaig** [comment]
thank you @igor-sirotin :) hope i will find someone to look into this

**Comment by anastasiyaig** [comment]
@borismelnik do u have an idea what could impact logging out?

**Comment by anastasiyaig** [comment]
perhaps this is the same when doing Change password

![image](https://github.com/status-im/status-desktop/assets/82375995/f8164067-61aa-43da-865a-09bae32ecd15)

Re-hashing DB for new account (empty DB) takes sometimes way too long

**Comment by iurimatias** [comment]
@anastasiyaig what version was this?

**Comment by anastasiyaig** [comment]
@iurimatias it is nightly usually, can't remember which exactly, but it happens in nightly runs quite often

**Comment by anastasiyaig** [comment]
@iurimatias so for example this happened in https://ci.status.im/job/status-desktop/job/e2e/job/prs/1092/console

change password test total execution time exceeded timeframe of 180 seconds (that i specifically set as this). Indirectly it could be said that the logging out took more than 2 minutes. And tests are using brand new account each time, so DB is tiny

**Comment by anastasiyaig** [comment]
so it now takes more than 15 seconds sometimes to rehash. With autotest having tiny DB i guess it is not really expected

**Comment by jrainville** [comment]
> @jrainville maybe you have insight of the root cause of this? I can constantly reproduce it today

@anastasiyaig I'm fairly sure this is because of the store node requests. We have them with a timeout of 10-20 seconds, so if we requested (in the background) messages from the store nodes just before the log out, we need to wait for them to time out or return, because the Nim app opened threads that are waiting for them.

Those timeouts used to be like 2 minutes or even had no timeouts, so that was the issue where the app would hang trying to close.

I'm not sure if there is much we can do. Maybe there is some sort of signal we can send to status-go to kill all requests, but it might be complex, I'm not sure.

**Comment by anastasiyaig** [comment]
i am wondering if i use brand new account (0 messages at all) - what to do with store nodes?
also, just wonder what is approximate logout time i can set for test to wait, right now it is 10 seconds, what is acceptable for fresh account (which autotest is using )? like 20 seconds?
Same for changing password @jrainville

**Comment by anastasiyaig** [comment]
thats what i expect as well, but it is more than 15 seconds now in both cases for brand new accounts

**Comment by richard-ramos** [comment]
> If we are not subscribed to any chat, I think it wouldn't so any store node fetch, but maybe it still does it for trying to get back ups? 

Yeah, even if you're not subscribed to chats, there are still history requests going on some topics you subscribe automagically (like for receiving 1:1 requests) as well as backups.  The context should have been cancelled on logout, so any active or future query should have failed.

**Comment by anastasiyaig** [comment]
@jrainville i tested that with Qt6 app and the problem is still there. tests are failing because this screen locks the app

![Image](https://github.com/user-attachments/assets/fbd6d6bb-859e-4aef-9dc5-b41e08ab9996)
